storage/http_client_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCreateBucketEmulated_HTTP(t *testing.T) {
+	if os.Getenv("STORAGE_EMULATOR_HOST") == "" {
+		t.Skip("Emulator tests skipped without emulator running")
+	}
+
+	c, err := newHTTPStorageClient(context.Background())
```

Comment:
So, the test code is basically identical and both constructors return the interface... can we just write once and run twice in a loop, swapping out the constructor?

Reply:
Yeah, we can do this. Should the tests be added to `client_test.go` instead? @BrennaEpp WDYT?

The only concern I have is if future tests/methods will have transport-specific things we want to test does it make sense to have two files?

Reply:
Ok I combined the tests in `client_test.go` and integrated an `initEmulatedClients` helper with the `TestMain` in `integration_test.go`. Works well.